app/gui/src/project-view/components/MarkdownEditor/htmlToMarkdown/htmlToMarkdownImpl.ts

```diff
@@ -0,0 +1,21 @@
+import { IMAGE } from '@/components/MarkdownEditor/htmlToMarkdown/imageNode'
+import { LINK } from '@/components/MarkdownEditor/htmlToMarkdown/linkNode'
+import { $generateNodesFromDOM } from '@lexical/html'
+import { $convertToMarkdownString, TRANSFORMERS } from '@lexical/markdown'
+import { $insertNodes, createEditor } from 'lexical'
+
+const domParser = new DOMParser()
+
+const transformers = [IMAGE, LINK, ...TRANSFORMERS]
+
+/** Convert the given HTML text to a Markdown approximation. */
+export function htmlToMarkdownImpl(html: string): string {
+  const dom = domParser.parseFromString(html, 'text/html')
+  const editor = createEditor({
+    nodes: transformers.flatMap((transformer) =>
+      'dependencies' in transformer ? transformer.dependencies : [],
+    ),
+  })
+  editor.update(() => $insertNodes($generateNodesFromDOM(editor, dom)))
```

Comment:
I’m not sure how realistic this is, but could we avoid creating the editor instance here, and serialize generated nodes directly to markdown representation?

Reply:
Not easily. After we do the main performance optimization (taking just the code we need from the various rich text packages), we could examine that.